types/architecture.pp

```diff
@@ -0,0 +1,9 @@
+type Poudriere::Architecture = Enum[
```

Comment:
can you please add puppet-strings style documentation for the new datatype?

Reply:
Done!